### ISSUE TITLE
Sandbox raises error on shutdown

### ISSUE THREAD

**Issue by letmaik** [microsoft/CCF#3658]
**Describe the bug**
```
^C15:11:12.242 | INFO     | __main__:run:117 - Stopping all CCF nodes...
15:11:12.242 | INFO     | infra.network:network:1349 - Stopping network
15:11:12.242 | INFO     | infra.node:stop:408 - Stopping node 0
15:11:12.243 | INFO     | infra.remote:stop:510 - [127.0.0.1] closing
15:11:12.306 | ERROR    | infra.remote:stop:517 - 127.0.0.1 exited with signal: Interrupt
15:11:12.308 | INFO     | infra.network:stop_all_nodes:662 - All nodes stopped
Traceback (most recent call last):
  File "/opt/ccf/bin/start_network.py", line 166, in <module>
    run(args)
  File "/opt/ccf/bin/start_network.py", line 117, in run
    LOG.info("Stopping all CCF nodes...")
  File "/usr/lib/python3.8/contextlib.py", line 120, in __exit__
    next(self.gen)
  File "/opt/ccf/bin/infra/network.py", line 1350, in network
    net.stop_all_nodes(skip_verification=True)
  File "/opt/ccf/bin/infra/network.py", line 663, in stop_all_nodes
    self.check_ledger_files_identical(**kwargs)
  File "/opt/ccf/bin/infra/network.py", line 634, in check_ledger_files_identical
    f"Verified {len(longest_ledger_files)} ledger files consistency on all {len(self.nodes)} stopped nodes"
TypeError: object of type 'NoneType' has no len()
```

**To Reproduce**
Start the sandbox and terminate it again.

**Environment information**
CCF 2.0.0 rc4

**Comment by jumaffre** [comment]
Is this because no package is specified to the sandbox script, i.e. `$ ../tests/sandbox/sandbox.sh # no arguments`? If so, we could simply guard the `Verified...` log line with `if longest_ledger_files:`.

**Comment by letmaik** [comment]
No, I think it's because I shut it down before the first file is written, or something like that.

**Comment by jumaffre** [comment]
I've fixed this in #3659. Does the sandbox still raise an error with this fix in? I'd expect it does, otherwise the nodes should start successfully and produce at least a committed ledger file.

**Comment by letmaik** [comment]
> I've fixed this in #3659. Does the sandbox still raise an error with this fix in? I'd expect it does, otherwise the nodes should start successfully and produce at least a committed ledger file.

Yes, that works. Thanks!

**Comment by eddyashton** [comment]
Resolved in #3659.
